kubernetes/resource_kubernetes_ingress.go

```diff
@@ -314,6 +314,22 @@ func resourceKubernetesIngressDelete(ctx context.Context, d *schema.ResourceData
 		return diag.Errorf("Failed to delete Ingress %s because: %s", d.Id(), err)
 	}
 
+	err = resource.RetryContext(ctx, d.Timeout(schema.TimeoutDelete), func() *resource.RetryError {
+		_, err := conn.ExtensionsV1beta1().Ingresses(namespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			if statusErr, ok := err.(*errors.StatusError); ok && statusErr.ErrStatus.Code == 404 {
```

Comment:
The `apimachinery` module contains a range of helper function that make checking the errors returned by the API quite easy and clean. It would be nice to make use of these for the sake of legibility and to avoid depending on implementation details or the err type.

In this case, I would this function:
https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go#L480

Reply:
Ok, I also replaced the comparison with 404 with that helper function `errors.IsNotFound()` in other files